cirq/ops/phased_x_gate.py

```diff
@@ -135,13 +162,36 @@ def __str__(self):
         return '{}^{}'.format(info.wire_symbols[0], info.exponent)
 
     def __repr__(self):
-        return 'cirq.PhasedXPowGate({!r}, {!r})'.format(self._phase_exponent,
-                                                        self._exponent)
+        args = ['phase_exponent={!r}'.format(self.phase_exponent)]
+        if self.exponent != 1:
+            args.append('exponent={!r}'.format(self.exponent))
+        if self._global_shift != 0:
+            args.append('global_shift={!r}'.format(self._global_shift))
+        return 'cirq.PhasedXPowGate({})'.format(', '.join(args))
+
+    def _period(self):
+        exponents = [self._global_shift, 1 + self._global_shift]
+        real_periods = [abs(2/e) for e in exponents if e != 0]
+        int_periods = [int(np.round(e)) for e in real_periods]
+        if any(i != r for i, r in zip(real_periods, int_periods)):
+            return None
+        if len(int_periods) == 1:
+            return int_periods[0]
+        return int_periods[0] * int_periods[1] / fractions.gcd(*int_periods)
```

Comment:
fractions.gcd is deprecated, I think you can use math.gcd?

Reply:
`math.gcd` doesn't exist in python 2, so we're stuck with `fractions.gcd` for now. We could consider modifying the transpilation script to handle this case.